File: app/src/main/java/com/ruuvi/station/database/tables/Alarm.java
package com.ruuvi.station.database.tables;

import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.PrimaryKey;
import com.raizlabs.android.dbflow.annotation.Table;
import com.raizlabs.android.dbflow.sql.language.SQLite;
import com.raizlabs.android.dbflow.structure.BaseModel;
import com.ruuvi.station.database.LocalDatabase;

import java.util.Date;
import java.util.List;

/**
 * Created by tmakinen on 26.7.2017.
 */

@Table(database = LocalDatabase.class)
public class Alarm extends BaseModel {
    public static final int TEMPERATURE = 0;
    public static final int HUMIDITY = 1;
    public static final int PRESSURE = 2;
    public static final int RSSI = 3;
    public static final int MOVEMENT = 4;

    @Column
    @PrimaryKey(autoincrement = true)
    public int id;
    @Column
    public String ruuviTagId;
    @Column
    public int low;
    @Column
    public int high;
    @Column
    public int type;
    @Column
    public boolean enabled;
    @Column
    public Date mutedTill;

    public Alarm() {
    }

    public Alarm(int low, int high, int type, String tagId) {
        this.enabled = true;
        this.low = low;
        this.high = high;
        this.type = type;
        this.ruuviTagId = tagId;
    }

    public static List<Alarm> getAll() {
        return SQLite.select()
                .from(Alarm.class)
                .queryList();
    }

    public static List<Alarm> getForTag(String id) {
        return SQLite.select()
                .from(Alarm.class)
                .where(Alarm_Table.ruuviTagId.eq(id))
                .queryList();
    }

    public static Alarm get(int id) {
        return SQLite.select()
                .from(Alarm.class)
                .where(Alarm_Table.id.eq(id))
                .querySingle();
    }
}

File: app/src/main/java/com/ruuvi/station/database/tables/Alarm.kt
package com.ruuvi.station.database.tables

import com.raizlabs.android.dbflow.annotation.Column
import com.raizlabs.android.dbflow.annotation.PrimaryKey
import com.raizlabs.android.dbflow.annotation.Table
import com.ruuvi.station.alarm.domain.AlarmType
import com.ruuvi.station.database.domain.LocalDatabase
import java.util.*

@Table(
    database = LocalDatabase::class,
    useBooleanGetterSetters = false
)
data class Alarm (
    @Column
    @PrimaryKey(autoincrement = true)
    var id: Int = 0,
    @Column
    var ruuviTagId: String = "",
    @Column
    var low: Int = 0, // Obsolete
    @Column
    var high: Int = 0, // Obsolete
    @Column
    var min: Double = 0.0,
    @Column
    var max: Double = 0.0,
    @Column
    var type: Int = 0,
    @Column
    var enabled: Boolean = false,
    @Column
    var mutedTill: Date? = null,
    @Column
    var customDescription: String = ""
) {
    val alarmType: AlarmType
        get() = AlarmType.getByDbCode(type)

    companion object {
        const val TEMPERATURE = 0
        const val HUMIDITY = 1
        const val PRESSURE = 2
        const val RSSI = 3
        const val MOVEMENT = 4
    }
}